libs/design/src/atoms/progress-indicator/progress-indicator.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnInit, Input, ChangeDetectionStrategy, ElementRef, Output, EventEmitter } from '@angular/core';
+
+import { daffColorMixin, DaffColorable, DaffPalette } from '../../core/colorable/colorable';
+import { daffProgressIndicatorAnimation } from './animation/progress-indicator-animation';
+import { AnimationEvent } from '@angular/animations';
+
+/**
+ * An _elementRef is needed for the Colorable mixin
+ */
+export class DaffProgressIndicatorBase{
+  constructor(public _elementRef: ElementRef) {}
+}
+
+const _daffProgressIndicatorBase = daffColorMixin(DaffProgressIndicatorBase, 'primary') 
+
+@Component({
+  selector: 'daff-progress-indicator',
+  templateUrl: './progress-indicator.component.html',
+  styleUrls: ['./progress-indicator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    daffProgressIndicatorAnimation.fill
+  ]
+})
+export class DaffProgressIndicatorComponent extends _daffProgressIndicatorBase implements DaffColorable{
+
+  /**
+   * The color of the progress indicator
+   * See DaffColorable
+   */
+  @Input() color: DaffPalette;
+
+  /**
+   * The percentage completion of the progression, 
+   * expressed as a whole number between 0 and 100.
+   * 
+   */
+  @Input() percentage = 0;
+
+  /**
+   * An event that emits each time the progression reaches 100% 
+   * and the animation is finished
+   */
+  @Output() finished: EventEmitter<void> = new EventEmitter();
+
+
+  /**
+   * Calculates when the progress animation is fully completed
+   * @param event: AnimationEvent
+   */
+  onAnimationComplete(event: AnimationEvent) : void {
+    // @ts-ignore: @angular/animations typing error on event.toState as string
+    // See: https://github.com/angular/angular/issues/26507
+    if(event.toState === "100" || event.toState === 100) {
```

Comment:
So this doesn't work, because of the github issue?
```
if(event.toState.toString() === "100")
```

Reply:
So, it does and doesn't. I have to turn off typing because:

```
AnimationEvent.toState: string;
```

When in actuality, it could be `number` or `boolean`. 

The Angular docs even have some info on doing `:increment`/`:decrement` short-hands on animations, so they're aware that it could potentially be numeric.

I have the two cases there because I don't know what they'll do in the end:

1. Coerce to string and keep the type
2. Allow other types.

Once that issue is resolved, we can remove the relevant other case.

Reply:
I'll add the [Github reference](https://github.com/angular/angular/issues/26507) here as well.